include/seastar/util/spinlock.hh

```diff
@@ -96,7 +96,9 @@ public:
     }
     void lock() noexcept {
         while (_busy.exchange(true, std::memory_order_acquire)) {
-            internal::cpu_relax();
+            while (_busy.load(std::memory_order_relaxed)) {
+                internal::cpu_relax();
+            }
```

Comment:
Patch looks okay, but did you see a problem? We use the spinlock very sparingly.

Also note: if there is no contention, then the new code is slower, since it will need two cache transactions (one to move to shared state, the other to move it to modified/exclusive state. But it's probably better overall.

Reply:
No, I haven't seen a problem. Maybe some (micro)benchmarks are needed (2 and 3 threads).

You are right, the hypothesis that new implementation slows down or speeds up the execution time, is needed to be proved by benchmark. I will write some (probably next week). My hypothesis is that `unlock` operation is affected by this busy `exchange` operation (because of MESI).

When many threads (>= 2) try to lock already locked spinlock, this patch improves execution time (reduces number of transactions in MESI).